### DISCUSSION TITLE
Understanding Login and signups for my application

### DISCUSSION THREAD

**Discussion by manojahi** [Budibase/budibase#3538] [Help]
Hi folks,

I tried Budibase and wanted to have the OAuth integrated for my apps where user can sign up and login for their content.
I see that in the admin side we can configure google OAuth to integrate the `login with google`, here are some questions i got.

- How do user sign up? I see only `login` option.
- Can i disable `custom login form e.g. with email and password` as i want to have only `login with google`
- Is there any documents for screen permissions, I see that `public` is for everyone. `Basic` is for user who logged in. `admin` is for users who have admin role, but what about `power`?
- I published my application with one screen and gave it as `basic` access for the screen. I then went to my app homepage which redirected to me `.../builder` and then to `login page`, so i logged in with google, but then i got redirected to `builder admin page` where i was able to see all the apps, I did not get why it did not go to my `application`. Am i missing something?

I have just started exploring `Budibase` and will have more questions as i go. thanks :)

**Comment by mjashanks** [comment]
> How do user sign up? I see only login option.

There is no way for users to self sign up - accounts must be created by an admin. We have not seen a huge need within internal apps for users to self-sign up.

> Can i disable custom login form e.g. with email and password as i want to have only login with google

No, It's a good idea, that has been raised before, please upvote! https://github.com/Budibase/budibase/discussions/2609

> Is there any documents for screen permissions, I see that public is for everyone. Basic is for user who logged in. admin is for users who have admin role, but what about power?

The capability of each role is defined in the app itself. You can
1. Set read/write access for a table, to a role
2. Allow screens to be seen only by certain roles
3. Use Conditions to change what is displayed in the app, based on roles
4. Add your own customer rows, outside of basic, admin and power

If you don't configure what the roles do, then they all have access to the same.

This is an area that definitely needs more documentation. Thanks for highlighting.

> I published my application with one screen and gave it as basic access for the screen. I then went to my app homepage  ...

I see - yes I would have expected the same thing - to go directly to the app after login. I will log this as an issue. Thanks!

**Reply by manojahi** [reply]
>There is no way for users to self sign up - accounts must be created by an admin. We have not seen a huge need within internal apps for users to self-sign up.

Oh, I was trying out budibase for creating consumer facing apps but don't think it can be used here.

**Reply by adecor-dev** [reply]
@mjashanks I'm also creating consumer facing apps. Can you confirm if Budibase is setup for this at the moment?

Why can't we set the access level at the time of onboarding a user via email?

**Reply by mjashanks** [reply]
@adecor-dev  There's no reason why you can't create externally facing apps with Budibase, so long as you are happy to manually create logins for all users. 

However, Budibase is not designed for creating your own SaaS product - there are a bunch of features that we will likely never provide, that would be important for this style of software.... self sign-up being one of those. Budibase is aimed at allowing organisations to create software for their own purposes (even if that means giving access to external users).

> Why can't we set the access level at the time of onboarding a user via email?

We just haven't implemented this - I can see why it would be a useful feature.

**Reply by nigel-daniels** [reply]
Hi, I also just started to look at Budibase as a tool to rapidly build a SaaS product, so reading this is very disappointing as Budibase is great in all other respects. Is there another low-code tool recommended for SaaS building?
